### ISSUE TITLE
[Bug] Text highlights and text cursor not visible when using Windows OS-level high-contrast themes

### ISSUE THREAD

**Issue by jruales** [microsoft/monaco-editor#4960]
### Reproducible in vscode.dev or in VS Code Desktop?

- [x] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Link

The default playground page reproes the issue:
https://microsoft.github.io/monaco-editor/playground.html

The Monaco Editor homepage also reproes the issue:
https://microsoft.github.io/monaco-editor/

The issue happens even in older versions. It doesn't seem like a regression.

### Monaco Editor Playground Code

```typescript
N/A (code-independent)
```

### Reproduction Steps

1. In the Windows OS settings, go to Accessibility -> Contrast themes. Then select any of the high-contrast themes from the dropdown and click Apply.
    <img width="1460" height="1474" alt="Image" src="https://github.com/user-attachments/assets/8ac2b2a9-af29-4ae0-948f-faf99a327e16" />
2. Open [Monaco Playground](https://microsoft.github.io/monaco-editor/playground.html) or [Monaco Homepage](https://microsoft.github.io/monaco-editor/). Reload the page just in case.

### Actual (Problematic) Behavior

1. Type in one of the editors, and notice that the text cursor is invisible (it's the same color as the background)
2. Try to select some text and notice that the text selection is invisible (it doesn't seem like the text is selected, even when it is)

![Image](https://github.com/user-attachments/assets/dbd9347f-16a0-441d-97f9-cd83d9db870b)

### Expected Behavior

The text cursor and the text selection should be visible

### Additional Context

The solution will likely use `@media ( forced-colors: active ) { … }` and `forced-color-adjust: none;`.

For fixing the cursor issue, it seems like the following might suffice:

```scss
  // OS-level high-contrast mode
  @media (forced-colors: active) {
    // Fix issue where text cursor was invisible (it was the same color as the background)
    .monaco-editor .cursors-layer>.cursor {
      // This prevents OS-level high-contrast mode from overriding the style of this element
      forced-color-adjust: none;
      // Note: CanvasText is a system color: https://www.w3.org/TR/css-color-4/?utm_source=chatgpt.com#css-system-colors
      background-color: CanvasText !important;
      border-color: CanvasText !important;
    }
  }
```

For the other issue, the highlight issue, maybe something like this would work:
(note on the below selectors: `.inline-selected-text` only seems to exist when the monaco theme is set to a high-contrast theme)
```scss
  // OS-level high-contrast mode
  @media (forced-colors: active) {
    // Fix issue where text selection was invisible (it was the same color as the background)
    // Note: the `.inline-selected-text` element only exists when using Monaco high-contrast themes.
    .monaco-editor .inline-selected-text {
      // This prevents OS-level high-contrast mode from overriding the style of this element
      forced-color-adjust: none;
      // Note: HighlightText and Highlight are system colors: https://www.w3.org/TR/css-color-4/?utm_source=chatgpt.com#css-system-colors
      color: HighlightText !important;
      background: Highlight !important;
    }
  }
```

Another option is to target `.selected-text` instead of `inline-selected-text`, so that this works for non-high-contrast Monaco themes as well, but that would be more involved, because the the OS also seems to force background color below all text, which would mean that the editor text would end up being forced to have a background, which would cove then normal selection rectangle element (which we want to be seen, not occluded). So it would be necessary to do `forced-color-adjust: none` overrides also to other elements above.

**Comment by jruales** [comment]
Looking at other existing bugs, it seems like this might be a partial duplicate of #4945 for the selection. Although this bug also covers the invisible cursor, which isn't covered by that other bug, and this bug has more details and possible solutions.

**Comment by adriano56789** [comment]
Estou entrando em contato para relatar um problema grave de segurança e interferência externa no Editor Monaco que estou utilizando no iStudio. Durante minhas análises, identifiquei que:

Códigos de terceiros estão sendo injetados no editor, causando:

Apagamento de APIs próprias do meu projeto.

Lentidão e travamento do site/editor.

Comportamento inesperado ao executar comandos ou gerar telas, criando saídas diferentes do solicitado.

Possível origem: scripts injetados por terceiros (Webmasters ou fornecedores de internet), o que caracteriza interferência externa deliberada.

Impacto:

Interrupção do desenvolvimento e testes de aplicativos.

Risco de perda de dados e código.

Comprometimento da integridade e confiabilidade do editor.

Diante disso, solicito que este problema seja investigado formalmente, pois trata-se de uma invasão de software e potencial crime digital.
